lib/srv/alpnproxy/proxy_test.go

```diff
@@ -398,9 +399,14 @@ func TestProxyMakeConnectionHandler(t *testing.T) {
 	require.Equal(t, string(common.ProtocolHTTP), clientTLSConn.ConnectionState().NegotiatedProtocol)
 	require.NoError(t, req.Write(clientTLSConn))
 
-	response, err := http.ReadResponse(bufio.NewReader(clientTLSConn), req)
+	resp, err := http.ReadResponse(bufio.NewReader(clientTLSConn), req)
 	require.NoError(t, err)
-	require.Equal(t, http.StatusOK, response.StatusCode)
+
+	// Always drain/close the body.
+	io.Copy(io.Discard, resp.Body)
+	_ = resp.Body.Close()
```

Comment:
If this is a test, shouldn't we assert the result here?

Reply:
I don't think the test cares about body.Close errors if everything else works.